app/Components/Game/index.js

```diff
@@ -2,64 +2,71 @@ import React from 'react';
 import ChatApp from '../ChatApp';
 import Profile from '../Profile';
 import Quiz from '../Quiz';
-
+import RoleSelectionMenu from '../RoleSelectionMenu';
 import io from 'socket.io-client';
 let socket = io.connect('');
  
 class Game extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {at_capacity: false, role: null, round_over: false, is_active_game: false};
-    socket.on('connect', () => this._connect());
-    socket.on('init', (username) =>this._initialize(username));
-    socket.on('getroom', () => this._get_room());
-    socket.on('adduser', (user_role) => this._add_user(user_role));
-    socket.on('fullhouse', () => this._full_house());
-    socket.on('isgameID', (flag) => this._is_game_ID(flag));
+    this.state = {at_capacity: false, role: null, round_over: false, is_active_game: false, takenRoles: null};
+    socket.on('isgameID', (flag, takenRoles) => this._is_game_ID(flag, takenRoles));
+    this.setRoleOptions = this.setRoleOptions.bind(this);
+    this.selectRole = this.selectRole.bind(this);
   }
 
   componentDidMount() {
-    // socket.emit('setroom', this.props.params.gameID);
     socket.emit('joingame', this.props.params.gameID);
   }
 
-  _connect() {
-    socket.emit('setroom', this.props.params.gameID);
-  }
-
-  _initialize(username) {
-    this.setState({user: username});
+  setRoleOptions(takenRoles) {
```

Comment:
This method name sounds like it's making a call to `setState` but it's really returning a description of the UI like a render method.  I'd suggest something like `renderRoleOptions` to indicate that.  The `eslint` rules can also check on things like that too.